hyppo/ksample/hhg.py

```diff
@@ -0,0 +1,128 @@
+    
+    Then apply the KS 2-sample test on these center-point distances. This classic test
+    compares the empirical distribution function of the two samples and takes
+    the supremum of the difference between them. See Notes under scipy.stats.ks_2samp
+    for more details.
+    
+    To achieve better power, the above process is repeated with each sample point 
+    :math:`x_i` and :math:`y_i` as center points. The resultant :math:`n+m` p-values
+    are then pooled for use in the Bonferroni test of the global null hypothesis.
+    The HHG statistic is the KS stat associated with the smallest p-value from the pool,
+    while the HHG p-value is the smallest p-value multipled by the number of sample points.
+    
+    References
+    ----------
+    .. footbibliography::
+    """
+
+    def __init__(self, compute_distance="euclidean", **kwargs):
+        self.compute_distance = compute_distance
+        KSampleTest.__init__(self, compute_distance=compute_distance, **kwargs)
+
+    def statistic(self, x, y):
+        """
+        x,y : ndarray of float
+            Input data matrices. ``x`` and ``y`` must have the same number of
+            dimensions. That is, the shapes must be ``(n, p)`` and ``(m, p)`` where
+            `n` and  are the number of samples and `p` is the number of
+            dimensions.
+            
+        Returns
+        -------
+        stat : float
+            The computed KS test statistic associated with the lowest p-value.
+        """
+        xy = np.concatenate((x, y), axis=0)
+        distxy = _centerpoint_dist(xy, self.compute_distance, 1)
+        distx = distxy[:, 0 : len(x)]
+        disty = distxy[:, len(x) : len(x) + len(y)]
+        stats, pvalues = _distance_score(distx, disty)
+        minP = min(pvalues)
+        minstat = stats[pvalues.index(minP)]
+        self.stat = minstat
+        return self.stat
+
+    def test(self, x, y):
+        """
+        x,y : ndarray of float
+            Input data matrices. ``y1`` and ``y2`` must have the same number of
+            dimensions. That is, the shapes must be ``(n, p)`` and ``(m, p)`` where
+            `n` and `m` are the number of samples and `p` is the number of
+            dimensions.
+            
+        Returns
+        -------
+        stat : float
+            The computed KS test statistic associated with the lowest p-value.
+        pvalue : float
+            The computed HHG pvalue. Equivalent to the lowest p-value multiplied by the total number
+            of samples.
+        """
+        check_input = _CheckInputs(inputs=[x, y],)
+        x, y = check_input()
+
+        xy = np.concatenate((x, y), axis=0)
+        distxy = _centerpoint_dist(xy, self.compute_distance, 1)
+        distx = distxy[:, 0 : len(x)]
+        disty = distxy[:, len(x) : len(x) + len(x)]
+        stats, pvalues = _distance_score(distx, disty)
+        minP = min(pvalues)
+        minstat = stats[pvalues.index(minP)]
+        pvalue = minP * len(pvalues)
```

Comment:
lines 94-100 are the same as those in `statistic` unless I am missing something. Call the `self.statistic` method instead

Reply:
Okay, changed that.